Mathlib/Data/Seq/Seq.lean

```diff
@@ -241,6 +249,44 @@ def recOn {motive : Seq α → Sort v} (s : Seq α) (nil : motive nil)
     rw [destruct_eq_cons H]
     apply cons
 
+@[simp]
+theorem noConfusion {x : α} {s : Seq α} : (cons x s) ≠ .nil := by
+  intro h
+  apply_fun head at h
+  simp at h
+
+@[simp]
+theorem noConfusion_symm {x : α} {s : Seq α} : .nil ≠ (cons x s) := noConfusion.symm
+
+theorem cons_eq_cons {x x' : α} {s s' : Seq α} :
+    (cons x s = cons x' s') ↔ (x = x' ∧ s = s') := by
+  constructor
+  · intro h
+    constructor
+    · apply_fun head at h
+      simpa using h
+    · apply_fun tail at h
+      simpa using h
+  · intro ⟨hx, hs⟩
+    congr
```

Comment:
This also works, use whichever you like most.
```suggestion
  · rintro ⟨rfl, rfl⟩
    rfl
```

Reply:
I prefer `intro` wherever it works. Are there any benefits to using `rintro`?